src/Evolution/Systems/NewtonianEuler/Limiters/KxrcfTci.hpp

```diff
@@ -0,0 +1,201 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/SliceVariables.hpp"
+#include "DataStructures/Tags.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Slice.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/SizeOfElement.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Tags.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp"
+#include "NumericalAlgorithms/LinearOperators/MeanValue.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+
+/// \cond
+template <size_t VolumeDim>
+class ElementId;
+/// \endcond
+
+namespace NewtonianEuler {
+namespace Limiters {
+namespace Tci {
+
+/// \ingroup LimitersGroup
+/// \brief Implements the troubled-cell indicator from Krivodonova et al, 2004.
+///
+/// The KXRCF (these are the author initials) TCI is described in
+/// \cite Krivodonova2004.
+///
+/// In summary, this TCI uses the size of discontinuities between neighboring DG
+/// elements to determine the smoothness of the solution. This works because the
+/// discontinuities converge rapidly for smooth solutions, therefore a large
+/// discontinuity suggests a lack of smoothness and the need to apply a limiter.
+///
+/// This implementation
+/// - does not support h- or p-refinement; this is checked by assertion.
+/// - chooses not to check external boundaries, because this adds complexity.
+///   However, by not checking external boundaries, the implementation may not
+///   be robust for problems that feed in shocks through boundary conditions.
+template <size_t VolumeDim, typename PackagedData>
+bool kxrcf_indicator(
+    const double kxrcf_constant, const Scalar<DataVector>& cons_mass_density,
```

Comment:
any guidance on choosing this constant that colud be added to the dox? It would be nice not to have to find the paper to see what some reference values would be.

Reply:
Good suggestion, I'll add a note!